gateway/ln-gateway/proto/gatewaylnrpc.proto

```diff
@@ -8,9 +8,11 @@ service GatewayLightning {
   /* GetPubKey returns the public key of the associated lightning node */
   rpc GetPubKey(GetPubKeyRequest) returns (GetPubKeyResponse) {}
 
-  /* PayInvoice attempts to pay an invoice using the associated lightning node
+  /* PayInvoice allows a client to stream requests to to pay any number of
+   * invoices using the associated lightning node
    */
-  rpc PayInvoice(PayInvoiceRequest) returns (PayInvoiceResponse) {}
+  rpc PayInvoice(stream PayInvoiceRequest) returns (stream PayInvoiceResponse) {
```

Comment:
Side note: I think the contract of this function needs to enforce idempotency or at least prevent double payments (re-tries if previous attempt failed are ok imo). If we receive a pay response and then crash and send the same pay request again on restart this should just return the same preimage again if payment succeeded previously.

Reply:
Agreed. We'd need service statefulness for that which is okay.
Aside from service crash and restarts, liquidity concerns might easily trigger the situation where this service needs to pause or save a failed pay invoice request, re-balance channels and then proceed / retry previously saved pay invoice requests